nntrainer/layers/cl_layers/rmsnorm_layer_cl.cpp

```diff
@@ -0,0 +1,397 @@
+
+    ret = context.command_queue_inst_.DispatchCommand(
+      RMSNormLayerCl::kernel_rmsnorm, work_groups_count, work_group_size);
+    if (!ret) {
+      break;
+    }
+
+    ret = resultbuf.ReadData(context.command_queue_inst_, rdata);
+    if (!ret) {
+      break;
+    }
+
+  } while (false);
+
+}
+
+void RMSNormLayerCl::rmsnormProcess_fp16(Tensor const &input,
+                                         Tensor &result,Tensor const &gamma,const float epsilon,
+                                         RunLayerContext &context){
+
+
+  bool ret = false;
+  int dim1 = input.batch() * input.height() * input.width()* input.channel();
+  CREATE_IF_EMPTY_DIMS(result, input.batch(), input.channel(), input.height(),
+                         input.width(), input.getTensorType());
+  int b=input.batch();
+  int c = input.channel();
+  int h = input.height();
+  int w = input.width();
+  do {
+    ret =
+      context.clCreateKernel(rmsnorm_cl_kernel_fp16_, context.LayerKernel::RMSNORM_FP16,
+                             RMSNormLayerCl::kernel_rmsnorm_fp16);
+    if (!ret) {
+      break;
+    }
+    opencl::Buffer inputbuf(context.context_inst_, dim1 * sizeof(cl_half), true,
+                          nullptr);
+
+    opencl::Buffer gammabuf(context.context_inst_, input.width() * sizeof(cl_half), true,
+                          nullptr);
+    opencl::Buffer resultbuf(context.context_inst_, dim1 * sizeof(cl_half), true, nullptr);
+
+    const __fp16 *data = input.getData<__fp16>();
+    __fp16 *rdata = result.getData<__fp16>();
+    const __fp16 *gdata = gamma.getData<__fp16>();
+    ret = inputbuf.WriteData(context.command_queue_inst_, data);
+    if (!ret) {
+      break;
+    }
+
+    ret = gammabuf.WriteData(context.command_queue_inst_, gdata);
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      0, &inputbuf, sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      1, &resultbuf, sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      2, &gammabuf, sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      4, &b, sizeof(int));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      3, &epsilon, sizeof(cl_half));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      5, &c, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      6, &h, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      7, &w, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    const int work_groups_count[3] = {1, 1, 1};
```

Comment:
{1, 1, 1} will have only one work item on the GPU.

```cpp
int n = get_global_id(0);
int c = get_global_id(1);
int h = get_global_id(2);
```
Above lines in the kernels will have `n = 0, c = 0, h = 0` always.  Please check this once using `print` statements or different input sizes.